### PR TITLE
CC-491: Consolidate and simplify unit tests of HDFS connector

### PR DESCRIPTION
**Pull request by kkonstantine**
Added a few test cases. 

But mainly adding necessary refactorings in junit tests in order to be able to write more tests and iterate easier over more records, in anticipation of additional formatters.

### CODE REVIEW

File: src/test/java/io/confluent/connect/hdfs/TestWithMiniDFSCluster.java [hunk 1]
```diff
@@ -69,4 +95,249 @@ private MiniDFSCluster createDFSCluster(Configuration conf) throws IOException {
...
+  /**
+   * Return a list of new records for a set of partitions, starting at the given offset in each partition.
+   *
+   * @param size the number of records to return.
+   * @param startOffset the starting offset.
+   * @param partitions the set of partitions to create records for.
+   * @return the list of records.
+   */
+  protected List<SinkRecord> createSinkRecords(int size, long startOffset, Set<TopicPartition> partitions) {
+    /*
```
Comment (ewencp) [inline src/test/java/io/confluent/connect/hdfs/TestWithMiniDFSCluster.java]: leftover code that could be cleaned up?

File: src/test/java/io/confluent/connect/hdfs/TestWithMiniDFSCluster.java [hunk 2]
```diff
@@ -69,4 +95,249 @@ private MiniDFSCluster createDFSCluster(Configuration conf) throws IOException {
...
+
+    List<SinkRecord> sinkRecords = new ArrayList<>();
+    for (long offset = startOffset; offset < startOffset + size; ++offset) {
+      sinkRecords.add(new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, key, schemaNoVersion,
+                                     recordNoVersion, offset));
+    }
+    return sinkRecords;
+  }
+
+  protected List<SinkRecord> createSinkRecordsWithAlteringSchemas(int size, long startOffset) {
```
Comment (ewencp) [inline src/test/java/io/confluent/connect/hdfs/TestWithMiniDFSCluster.java]: alternating?

File: src/test/java/io/confluent/connect/hdfs/parquet/HiveIntegrationParquetTest.java [hunk 3]
```diff
@@ -300,38 +275,11 @@ public void testHiveIntegrationTimeBasedPartitionerParquet() throws Exception {
     assertEquals(9, rows.length);
     for (int i = 0; i < rows.length; ++i) {
       String[] parts = HiveTestUtils.parseOutput(rows[i]);
-      for (int j = 0; j < expectedResult.get(i).length; ++j) {
-        assertEquals(expectedResult.get(i)[j], parts[j]);
+      int j = 0;
+      for (String expectedValue : expectedResults.get(i)) {
+        System.err.println("Exp: " + expectedValue + " Actual: " + parts[j]);
```
Comment (ewencp) [inline src/test/java/io/confluent/connect/hdfs/parquet/HiveIntegrationParquetTest.java]: leftover debugging? and the index doesn't match the following assert?

**Comment by kkonstantine** [comment]
ok to test

**Pull request merged**